### DISCUSSION TITLE
Copilot needs to be reauthorized every time it starts

### DISCUSSION THREAD

**Discussion by mauryaratan** [community/community#7597] [Copilot Conversations] [answered]
I'm using PHPStorm 2021.2.3 and v1.0.3 of Copilot. Looks like every time PHPStorm is started, I need to authorized Github again, suggestions don't show up otherwise.

Upon checking with a teammate I found out it works fine for him in PHPStorm. Is anyone else having this issue or have any clue how to fix it?

I tried removing and installing again but that didn't help. Thanks.

**Comment by Sect0R** [comment]
1. Did you use SSH to connect to Github?
2. Check that your Github account is associated with Jetbrains (Settings -> Version control -> Github)
3. What OS did you use?

If all right - try to restart PHPStorm.

**Comment by jansorg** [comment, marked as answer]
1.0.3 should've fixed a similar issue.

The VCS settings are not used by the Copilot plugin, it's keeping it's own settings.
If a restart isn't helping, could you enable logging by added `#com.github.copilot` at `Help > Diagnostic Tools > Debug log settings`, then restart and then post the log file, please? The log file is available at `Help > Show Log in ...`.

**Reply by SATOSHI66** [reply]
> 1.0.3 should've fixed a similar issue.
> 
> The VCS settings are not used by the Copilot plugin, it's keeping it's own settings. If a restart isn't helping, could you enable logging by added `#com.github.copilot` at `Help > Diagnostic Tools > Debug log settings`, then restart and then post the log file, please? The log file is available at `Help > Show Log in ...`.

**Comment by mauryaratan** [comment]
@jansorg I attached the logs here: https://gist.github.com/mauryaratan/7cc32a8ce0589794af7ab7167bfa0552

Not sure if it's related but for me `Settings > Version Control > Github` is also logged out for me every time I restart PHPStorm.

I tried:
- Removing account altogether then adding again
- Logging in via Access Token
- Logging in directly via Github

Every time I restart PHPStorm, Github shows me logged out. Same as Copilot needs to be authorized again.

**Comment by mauryaratan** [comment]
Actually, please ignore. I found out the problem was something else, nothing related to Copilot.

In `Appearance & Behavior > System Settings > Passwords`, it was set not to save any, hence it kept getting reset after every restart.

All good now 👍🏻

**Reply by jansorg** [reply]
Thanks for the pointer! We might have to check the setting and notify the user, I guess.

**Comment by sulfo** [comment]
Having identical issue, but my phpstorm is set to save passwords in native keychain.
I will try to enable logging and see if I get something use full.

**Reply by jansorg** [reply]
Please try the latest update, it has a fix and shouldn't "forget" the login. Please let us know if this is still the case for you with the latest update.

**Reply by sulfo** [reply]
Hey - it worked! Awesome 🤘

**Reply by fredrikbergqvist** [reply]
Worked for me as well after updating, thanks!

**Comment by luigimannoni** [comment]
I am facing the same problem on vscode, I have a github copilot login prompt every week or so.

**Reply by myMartek** [reply]
I also have this on VS Code everytime I open VS Code

**Reply by davecheney** [reply]
@luigimannoni @myMartek this is about phpstorm, not vscode. Please start a new discussion
